eng/common/TestResources/New-TestResources.ps1

```diff
@@ -428,6 +428,7 @@ try {
         $deploymentOutputs = @{
             "$($serviceDirectoryPrefix)CLIENT_ID" = $TestApplicationId;
             "$($serviceDirectoryPrefix)CLIENT_SECRET" = $TestApplicationSecret;
+            "$($serviceDirectoryPrefix)PRINCIPAL_ID" = $TestApplicationOid;
```

Comment:
1. This file is sync'd from the azure-sdk-tools repo. You have to make the change there. See https://github.com/Azure/azure-sdk-tools/blob/master/doc/common/common_engsys.md, @benbp can help you if you need (I'm OOF).
2. The OID isn't the principal ID. It's the tenant-mapped client ID, or "object ID". Client ID and principal ID are really the same thing here. I'd call this "CLIENT_OID" if we're going to make it generic; otherwise, make sure your ARM template declares a parameter like below (it's passed automatically) and you can output it in your own "outputs". This would be much easier/faster than making a generic change, especially if no one else may need it.

```json
        "testApplicationOid": {
            "type": "string",
            "defaultValue": "b3653439-8136-4cd5-aac3-2a9460871ca6",
            "metadata": {
                "description": "The client OID to grant access to test resources."
            }
        },
```

I'm actually a little surprised since the OID isn't often used directly but in ARM templates. Normally the client (principal) ID is looked up either by the client or the backend because OIDs aren't referred to directly as much. Is this needed?